cmake/checks/check_01_cxx_features.cmake

```diff
@@ -176,11 +176,33 @@ IF(NOT DEFINED DEAL_II_WITH_CXX17 OR DEAL_II_WITH_CXX17)
       "
       DEAL_II_HAVE_CXX17_IF_CONSTEXPR)
 
+    #
+    # Some compilers treat lambdas as constexpr functions
+    # when compiling with C++17 support. Consequently, they
+    # don't allow try-catch-blocks in lambdas.
```

Comment:
If this is really just a bug, then maybe say that it is a bug to make that part clear.